sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2236,26 +2250,32 @@ def _provision(self,
                 # provisioning), then we know the exact zone already.
                 handle.launched_resources = handle.launched_resources.copy(
                     zone=zones[0].name)
-            elif (task.num_nodes == 1 or
-                  handle.launched_resources.zone is not None):
-                # Query zone if the cluster has 1 node, or a zone is
-                # specifically requested for a multinode cluster.  Otherwise
-                # leave the zone field to None because head and worker nodes
-                # can be launched in different zones.
+            else:
                 get_zone_cmd = (
                     handle.launched_resources.cloud.get_zone_shell_cmd())
                 if get_zone_cmd is not None:
-                    returncode, stdout, stderr = self.run_on_head(
-                        handle, get_zone_cmd, require_outputs=True)
-                    subprocess_utils.handle_returncode(returncode,
-                                                       get_zone_cmd,
-                                                       'Failed to get zone',
-                                                       stderr=stderr,
-                                                       stream_logs=stream_logs)
-                    # zone will be checked during Resources cls
-                    # initialization.
-                    handle.launched_resources = handle.launched_resources.copy(
-                        zone=stdout.strip())
+                    ssh_credentials = backend_utils.ssh_credential_from_yaml(
+                        handle.cluster_yaml)
+                    runners = command_runner.SSHCommandRunner.make_runner_list(
+                        ip_list, **ssh_credentials)
+
+                    def _get_zone(runner):
+                        returncode, stdout, stderr = runner.run(
+                            get_zone_cmd, require_outputs=True)
+                        subprocess_utils.handle_returncode(
+                            returncode,
+                            get_zone_cmd,
+                            'Failed to get zone for ',
+                            stderr=stderr,
+                            stream_logs=stream_logs)
+                        return stdout.strip()
+
+                    zones = subprocess_utils.run_in_parallel(_get_zone, runners)
+                    if len(set(zones)) == 1:
+                        # zone will be checked during Resources cls
+                        # initialization.
+                        handle.launched_resources = (
```

Comment:
Add?: If head and worker nodes are launched in different zones (legacy clusters before #1700), leave the zone field to None.

Reply:
Oh, I meant how about we leave it as a comment, rather than a warning log? (Users need not know such details?)

Reply:
Ahh. Makes sense! Fixed. Thanks!